subsys/bluetooth/host/gatt.c

```diff
@@ -2542,20 +2531,40 @@ int bt_gatt_notify_cb(struct bt_conn *conn,
 int bt_gatt_notify_multiple(struct bt_conn *conn, uint16_t num_params,
 			    struct bt_gatt_notify_params *params)
 {
-	int i, ret;
-
 	__ASSERT(params, "invalid parameters\n");
-	__ASSERT(num_params, "invalid parameters\n");
-	__ASSERT(params->attr, "invalid parameters\n");
+	__ASSERT(num_params >= 2, "invalid parameters\n");
+	__ASSERT(params->attr || params->uuid, "invalid parameters\n");
+	for (uint16_t i = 1; i < num_params; i++) {
+		__ASSERT(params[0].func == params[i].func, "invalid parameters\n");
+		__ASSERT(memcmp(params[0].user_data, params[i].user_data, 1) == 0,
```

Comment:
Why are we doing a `memcmp` of the first byte here instead of a pointer comparison?